src/core/stream/stream_loader.ts

```diff
@@ -194,6 +201,58 @@ export default function StreamLoader({
     const stallingManager$ = StallingManager(mediaElement, clock$)
       .pipe(map(EVENTS.stalled));
 
+    /**
+     * Creates an observable which will refresh the manifest every
+     * "minimumUpdatePeriod". The minimum update period is refreshed
+     * each time a new manifest is downloaded.
+     * @param {Object} manifest
+     * @returns {Observable}
+     */
+    function recursivelyUpdateManifest(
+      manifestToUpdate: Manifest
+    ): Observable<IManifestUpdateEvent> {
+      const { minimumUpdatePeriod } = manifestToUpdate;
+
+      if (minimumUpdatePeriod != null && minimumUpdatePeriod > 0) {
+        const now = Date.now() / 1000;
+        const refreshDelay = Math.max(
+          0,
+          (minimumUpdatePeriod * 1000) - (now -  manifestToUpdate.loadedAt)
+        );
+        return observableTimer(refreshDelay).pipe(
+          mergeMap(() => {
+            return observableConcat(
+              refreshManifest(fetchManifest, manifestToUpdate),
+              recursivelyUpdateManifest(manifestToUpdate)
+            );
+          })
+        );
+      }
+      return EMPTY;
+    }
+
+    const bufferNeedsToRefreshManifest$ = buffers$.pipe(
+      filter((evt) => evt.type === "needs-manifest-refresh")
+    );
+
+    // The recursive refreshing logic is reset each time the buffer
+    // needs a manifest refresh.
+    const updateManifest$ =
+      observableMerge(
+        recursivelyUpdateManifest(manifest).pipe(
+          takeUntil(bufferNeedsToRefreshManifest$)
+        ),
+        bufferNeedsToRefreshManifest$.pipe(
+          exhaustMap(() => refreshManifest(fetchManifest, manifest)),
+          switchMap((evt) => {
+            return observableConcat(
+              observableOf(evt),
+              recursivelyUpdateManifest(manifest)
+            );
+          })
+        )
```

Comment:
I don't get why we have to do all this convoluted code.

Wouldn't be a subject restarting a timer, launched at each manifest update sufficient?

(btw the exhaustMap is not a requirement as `fetchManifest` is already throttled. It's the best between the `smthgMap` operators for it, just to say that you can do without).